File: teoria/clase3/clase3_Ruby/Villanos/villano.rb

module Villanos
  class Villano

    def initialize(nombre, esconditeS, fortuna, mayordomo)
      @nombre = nombre
      @escondite_secreto = esconditeS
      @fortuna = fortuna
      @mayordomo = mayordomo
      @planes = Array.new
    end

    attr_writer :fortuna

    def seguir_plan_prioritario
    end

    def primer_plan_activo
    end

  end
end

File: teoria/clase3/src/villanos/Villano.java
package villanos;

import esbirros.Mayordomo;
import java.util.ArrayList;

/**
 *
 * @author Lugli, Valentino Glauco
 */
public class Villano {
    
    private String nombre;
    String esconditeSecreto;
    private int fortuna;
    
    ArrayList<Plan> planes;
    Mayordomo mayordomo;
    
    public Villano(String nombre, String escondite, int fortuna, Mayordomo mayordoo)
    {
        this.nombre = nombre;
        this.esconditeSecreto = escondite;
        this.fortuna = fortuna;
        this.planes = null;
        this.mayordomo = mayordomo;
    }

    public void setFortuna(int cantidad)
    {
        this.fortuna = cantidad;
    }
    
    public boolean seguirPlanPrioritario()
    {
        throw new UnsupportedOperationException("No implementado");
    }
    
    public Plan primerPlanActivo()
    {
        return this.planes.get(0);
    }
}
